control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3155,6 +3177,27 @@ func (r *HostedControlPlaneReconciler) reconcileMachineConfigServerConfig(ctx co
 	return nil
 }
 
+func (r *HostedControlPlaneReconciler) reconcileManagedTrustedCABundle(ctx context.Context, trustedCABundle *corev1.ConfigMap, caBundleConfigMaps []client.ObjectKey) error {
+	caBundles := make([]string, len(caBundleConfigMaps))
+	for _, key := range caBundleConfigMaps {
+		cm := &corev1.ConfigMap{}
+		if err := r.Get(ctx, key, cm); err != nil {
+			return fmt.Errorf("failed to get configMap %s: %w", key.Name, err)
+		}
+		data, hasData := cm.Data[certs.UserCABundleMapKey]
+		if !hasData {
+			return fmt.Errorf("configMap %s must have a %s key", cm.Name, certs.UserCABundleMapKey)
```

Comment:
As the PR it's right now, this would just silently fail with no clear reason for a consumer.
Can we make this validation within e.g. `validateConfigAndClusterCapabilities` in HC